src/components/AnnotationProps.tsx

```diff
@@ -42,6 +42,37 @@ export default interface AnnotationProps {
    */
   onDeselect?: () => void;
 
+  /**
+   * Event fired with the annotation is drag start.
+   */
+  onDragStart?: () => void;
+
+  /**
+   * Event fired with the annotation is drag end.
+   */
+  onDragEnd?: () => void;
```

Comment:
The API proposed in this PR lets the user move the annotations, but does not provide a way to get the new position of the annotation. This looks like an oversight, as it makes the new API useless in most cases. This also makes the position of the marker differ from what is indicated by its parameters in some cases.

We should probably pass the new position of the marker as a parameter to `onDragEnd`, so that we are able to get this information (in a similar way to how `onRegionChangeEnd` works). What do you think? Would that work well with the use case you have in mind for draggable annotations?

A small suggestion for the future: it is considered a good practice to first submit an issue with the proposed API changes before creating a pull request when doing non-trivial changes like this one.

Reply:
This is an absolutely valid point. Also the event improvement you are talking about would be one of the next things I had in mind for further PRs.

I might also open some issues for some further changes I had in mind :) Thank you!